Arm/rover_arm_system.hpp

```diff
@@ -194,68 +186,100 @@ class RoverArmSystem : public sjsu::common::RoverSystem
     elbow_.SetPosition(angle);
   }
 
+  void MoveHand()
+  {
+    hand_.HandleHandMovement();
+  }
+
+  // TODO: implement different arm drive modes in this function with switch
+  // statements
   void HandleRoverMovement() override
   {
-    // TODO: implement different arm drive modes in this function with switch
-    // statements
     MoveRotunda(mc_data_.rotunda_angle);
     MoveShoulder(mc_data_.shoulder_angle);
     MoveElbow(mc_data_.elbow_angle);
-    hand_.HandleHandMovement();
+    MoveHand();
   }
 
   void HomeArm()
   {
-    UpdateRotundaAcceleration();
-    UpdateShoulderAcceleration();
+    UpdateAccelerations();
     HomeShoulder();
-    UpdateElbowAcceleration();
+    UpdateAccelerations();
     HomeElbow();
-    hand_.HomeWrist(float(rotunda_.GetOffsetAngle()));  // move all other homing
-                                                        // functions to private
+    UpdateAccelerations();
+    HomeHand();
   }
 
-  void HomeShoulder()
+ private:
+ //TODO: change the joint class to have its own acceleration member variable to remove duplications in code like this
+  void UpdateAccelerations()
+  {
+    rotunda_.GetAccelerometerData();
+    shoulder_.GetAccelerometerData();
+    elbow_.GetAccelerometerData();
+    wrist_.GetAccelerometerData();
+  }
+
+  float CalculateShoulderHomeAngle()
   {
     float home_angle = 0;
-    // TODO: std might have solution for this e.g. std::clamp or std::max?
-    ChangeIfZero(accelerations_.rotunda.x);
-    ChangeIfZero(accelerations_.rotunda.y);
-    ChangeIfZero(accelerations_.shoulder.x);
-    ChangeIfZero(accelerations_.shoulder.y);
 
-    float acceleration_x = accelerations_.rotunda.x + accelerations_.shoulder.x;
-    float acceleration_y = accelerations_.rotunda.y + accelerations_.shoulder.y;
+    float acceleration_x = rotunda_.acceleration_.x + shoulder_.acceleration_.x;
+    float acceleration_y = rotunda_.acceleration_.y + shoulder_.acceleration_.y;
 
     home_angle = float(atan(acceleration_y / acceleration_x));
+    return home_angle;
+  }
+
+  float CalculateUncorrectedElbowHomeAngle()
+  {
+    float acceleration_x = rotunda_.acceleration_.x + elbow_.acceleration_.x;
+    float acceleration_y = rotunda_.acceleration_.y + elbow_.acceleration_.y;
+    float angle_without_correction =
+        float(atan(acceleration_y / acceleration_x));
+    return angle_without_correction;
+  }
+
+  bool InSecondQuadrantOfGraph()
```

Comment:
```suggestion
  bool IsInSecondQuadrantOfGraph()
```
Maybe rename these to be more descriptive? What exactly is in second quadrant?